libminifi/test/unit/RegexUtilsTests.cpp

```diff
@@ -0,0 +1,59 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "../TestBase.h"
+#include "utils/RegexUtils.h"
+#include <string>
+#include <vector>
+
+using org::apache::nifi::minifi::utils::RegexUtils;
+
+TEST_CASE("TestRegexUtils::single_match", "[regex1]") {
+    std::string pat = "Speed limit 130 | Speed limit 80";
+    std::string rgx1 = "Speed limit ([0-9]+)";
+    std::vector<RegexUtils::Mode> mode = {RegexUtils::Mode::EXTENDED, RegexUtils::Mode::ICASE};
+    RegexUtils r1(rgx1, mode);
+    REQUIRE(r1.canMatch(pat));
+    auto ret = r1.getResult();
+    std::vector<std::string> ans;
+    ans.push_back("Speed limit 130");
+    ans.push_back("130");
+    REQUIRE(ans.size() == ret.size());
+    for (size_t i = 0; i < ans.size(); ++i) {
+        REQUIRE(ans[i] == ret[i]);
+    }
+    REQUIRE("| Speed limit 80" == r1.getSuffix());
+}
+
+TEST_CASE("TestRegexUtils::invalid_construction", "[regex2]") {
+  std::string pat = "Speed limit 130 | Speed limit 80";
+  std::string rgx1 = "Speed limit ([0-9]+)";
+  std::string rgx2 = "[Invalid)A(F)";
+  std::vector<RegexUtils::Mode> mode = {RegexUtils::Mode::EXTENDED, RegexUtils::Mode::ICASE};
+  RegexUtils r1(rgx1, mode);
+  try {
+    RegexUtils r2(rgx2, mode);
+    std::terminate();
+  } catch (const std::string &msg) {
+#if (__cplusplus > 201103L) || defined(_WIN32)
+    REQUIRE("The expression contained mismatched [ and ]." == msg);
+#else
+    REQUIRE("brackets ([ ]) not balanced" == msg);
+#endif
+  }
+}
```

Comment:
This tests are cool, however I would prefer to have some tests on edgecases, like one that verifies icase mode, check empty inputs (either the string or the regex itself).